src/DigitalTwin/CodeGeneratorCore.ts

```diff
@@ -580,65 +580,58 @@ export class CodeGeneratorCore {
       md5value = targetConfigItem.codeGeneratorLocation.ubuntuMd5;
     }
 
-    const loading = setInterval(() => {
-      channel.append('.');
-    }, 1000);
-
-    try {
-      // Download
-      utils.channelShowAndAppend(
+    // Download
+    utils.channelShowAndAppend(
+        channel,
+        `Step 1: Downloading ${DigitalTwinConstants.codeGenCli} v${
+            newVersion} package ...`);
+    const downloadOption: request
+        .OptionsWithUri = {method: 'GET', uri: packageUri, encoding: null};
+    const zipData = await request(downloadOption).promise();
+    const tempPath = path.join(os.tmpdir(), FileNames.iotworkbenchTempFolder);
+    const filePath = path.join(tempPath, `${md5value}.zip`);
+    fs.writeFileSync(filePath, zipData);
+    // clearInterval(loading);

```

Comment:
better to remove unused code here

Reply:
Good catch. Will remove it in next iteration

---
In reply to: [361902534](https://github.com/microsoft/vscode-iot-workbench/pull/894#discussion_r361902534) [](ancestors = 361902534)